common/include/pcl/pcl_macros.h

```diff
@@ -448,3 +448,30 @@ aligned_free(void* ptr)
 #else
   #define PCL_IF_CONSTEXPR(x) if (x)
 #endif
+
+// [[unlikely]] can be used on any conditional branch, but __builtin_expect is restricted to the evaluation point
+// This makes it quite difficult to create a single macro for switch and while/if
+/**
+ * @def PCL_CONDITION_UNLIKELY
+ * @brief marks a conditional branch as more likely than the other
+ * \code{.cpp}
+ * if PCL_CONDITION_UNLIKELY(x == 0) { return; } else { throw std::runtime_error("some error"); }
+ * //
+ * while PCL_CONDITION_UNLIKELY(is_complete) { continue; }  // busy wait, with minimal chances of waiting
```

Comment:
Maybe `while PCL_CONDITION_UNLIKELY(wait_for_result) { sleep(1); }` would be a bit clearer? Or did I misunderstand the example?

Reply:
No, you didn't